rust/lance-encoding/src/encodings/logical/primitive.rs

```diff
@@ -2757,6 +2763,7 @@ impl StructuralPrimitiveFieldScheduler {
         page_index: usize,
         _column_index: usize,
         decompressors: &dyn DecompressionStrategy,
+        cache_repetition_index: bool,
```

Comment:
This is probably ok for now but it seems like these parameters could start to creep if we get more.  Maybe it is easier to pass...

```suggestion
        config: &SchedulerDecoderConfig,
```

...or something like that (where `DecoderOptions` is a new sruct.